data/database.py

```diff
@@ -1,58 +1,177 @@
-
-            c.execute("CREATE TABLE livingspace_table (id TEXT PRIMARY KEY,\
-                                wants_accommodation TEXT, livingspace TEXT)")
-
-
-            for person in person_list:
-                office = person.office.name if person.office is not None \
-                                                                    else ""
-                c.execute("INSERT INTO person_table (id, name, designation, \
-                        office) VALUES ({}, {}, {}, {})".format(person.ID, \
-                                    person.name, person.designation, office))
-                if person.designation.lower() == "fellow":
-                    LivingSpace = person.livingspace.name if \
-                                    person.livingspace is not None else ""
-                    c.execute("INSERT INTO livingspace_table (id, \
-                            wants_accommodation, livingspace) VALUES \
-                                ({}, {}, {})".format(person.ID, \
-                                    person.wants_accommodation, LivingSpace))
-
+            self.run_migrations(c)
+            self.insert_rooms(rooms, c)
+            self.insert_people(person_list, c)
             conn.commit()
             conn.close()
+            return "The state has been successfully saved with %s.sqlite" \
+                   % db_name
 
-    def insert_rooms(self, room_list):
-        for room in room_list:
+    def insert_rooms(self, rooms, c):
+        """
+        This function executes the sql command insert to
+        save each room in the Database
+        """
+        for room in rooms:
             try:
                 room_type = "office" if isinstance(room, Office) \
                                                         else "livingspace"
                 c.execute("INSERT INTO room_table (name, type, capacity) \
-                                    VALUES ({}, {}, {})".format(room.name,\
-                                                room_type, room.total_space))
+                          VALUES ('{}', '{}', '{}')".format(
+                                room.name, room_type, room.total_space))
             except sqlite3.IntegrityError:
                 print('ERROR: failed to insert room {}'.format(room.name))
+
+    def insert_people(self, person_list, c):
+        """
+        This function executes the sql command insert to
+        save each person in the Database
+        """
+        for person in person_list:
+            office = person.office.name if person.office is not None else ""
+            c.execute(
+                "INSERT INTO person_table (id, name, designation, " +
+                "office) VALUES ('{}', '{}', '{}', '{}')".format(
+                    person.ID, person.name, person.designation, office))
+            if person.designation.lower() == "fellow":
+                livingSpace = person.livingspace.name if \
+                                person.livingspace is not None else ""
```

Comment:
this statement would imply that we don't the the `if` above in Line 61.. because `person.livingspace` can be None for both fellows and staff